### PR TITLE
msys2: Conan 2.0 compatibility tweaks

### PR DESCRIPTION
**Pull request by System-Arch**
Specify library name and version:  **msys2/cci.latest**

While this recipe was ported to use Conan 2.0 modules (see #12715), it did not appear to actually function with Conan 2.0-beta6 or develop2. This set of changes addresses the remaining compatibility issues that were encountered as described in #14667

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by System-Arch** [comment]
Hi @prince-chrismc.
I think msys2 might be a good candidate to add to #12888. With these changes, it should now build with Conan 2.0. Thanks

**Comment by jwillikers** [comment]
@System-Arch Why the `no_kill` option? I take it that's unrelated to Conan 2.0 support?

**Comment by System-Arch** [comment]
> @System-Arch Why the `no_kill` option? I take it that's unrelated to Conan 2.0 support?

Hi @jwillikers,

Correct. It turns out that my Conan 2.0 build automation uses msys2 (not from a Conan package) and the kill, which is really unnecessary in the context of building a Conan package, was killing the automation in the middle of the build process.

**Comment by uilianries** [comment]
I agree with @SSE4 . Also, I don't think managing process life is a task for Conan client, it should be part from the system itself. I don't it's a good idea adding that option.

**Comment by System-Arch** [comment]
> I agree with @SSE4 . Also, I don't think managing process life is a task for Conan client, it should be part from the system itself. I don't think is a good idea adding that option.

Hi @SSE4, Hi @uilianries,
I'm sorry but I am confused by your comments. On the one hand you say "I don't think managing process life is a task for Conan client", which I agree with. On the other hand, the existing Conan client is doing just that by killing off arbitrary processes. Are you suggesting that I take out the no_kill option and leave the "kill" code as it was, or are you (hopefully) recommending removing the "kill" code thus eliminating the need for the "no_kill" option? Thanks for clarifying.

**Comment by prince-chrismc** [comment]
You can remove objectdump, seems the hooks is broken

File: recipes/msys2/all/conanfile.py [hunk 1]
```diff
@@ -161,22 +163,24 @@ def package(self):
...
+        # See https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#kb-h013-default-package-layout
+        copy(self, "*", dst=os.path.join(self.package_folder, "bin", "msys64"), src=self._msys_dir, excludes=excludes)
         shutil.copytree(os.path.join(self._msys_dir, "usr", "share", "licenses"),
                         os.path.join(self.package_folder, "licenses"))
 
     def package_info(self):
         self.cpp_info.libdirs = []
         self.cpp_info.includedirs = []
         self.cpp_info.resdirs = []
+        self.cpp_info.system_libs = ["objdump"]
```
Comment (SpaceIm) [inline recipes/msys2/all/conanfile.py:181]:
```suggestion
```

Just ignore system libs hooks for this recipe.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/msys2/all/conanfile.py [hunk 2]
```diff
@@ -53,11 +46,13 @@ class MSYS2Conan(ConanFile):
         "exclude_files": ["ANY"],
         "packages": ["ANY"],
         "additional_packages": [None, "ANY"],
+        "no_kill": [True, False]
     }
     default_options = {
         "exclude_files": "*/link.exe",
         "packages": "base-devel,binutils,gcc",
         "additional_packages": None,
+        "no_kill": False,
```
Comment (SpaceIm) [inline recipes/msys2/all/conanfile.py:55]: this option should be removed from package id

File: recipes/msys2/all/conanfile.py [hunk 3]
```diff
@@ -68,9 +63,16 @@ def validate(self):
         if self.settings.arch != "x86_64":
             raise ConanInvalidConfiguration("Only Windows x64 supported")
 
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
```
Comment (SpaceIm) [inline recipes/msys2/all/conanfile.py:69]: It's useless and confusing. There is no compiler in settings.

File: recipes/msys2/all/conanfile.py [hunk 4]
```diff
@@ -1,21 +1,15 @@
...
-try:
-    import ctypes
-except ImportError:
-    pass
-except ValueError:
-    pass
-
-required_conan_version = ">=1.47.0"
-
+required_conan_version = ">=1.55.0"
```
Comment (SpaceIm) [inline recipes/msys2/all/conanfile.py:12]: min conan version is 1.53.0 with these unecessary rm_safe, and without it's 1.47.0

File: recipes/msys2/all/conanfile.py (hunk 2 above)
Comment (prince-chrismc) [reply recipes/msys2/all/conanfile.py:55]: Oops! I for this was not resolved 🙈

Comment (System-Arch) [reply recipes/msys2/all/conanfile.py:55]: > this option should be removed from package id

Hi @SpaceIm,

Good catch. Thanks

File: recipes/msys2/all/conanfile.py (hunk 3 above)
Comment (System-Arch) [reply recipes/msys2/all/conanfile.py:69]: Hi @SpaceIm,
I guess I am the one who is confused. Please see https://github.com/conan-io/hooks/issues/471

**Pull request merged**